### PR TITLE
Add fix quality metadata and `fixedFindings` for changeset entry

### PR DESCRIPTION
**Pull request by drdavella**

### CODE REVIEW

**Review by nahsra — APPROVED** [review]

File: src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java [hunk 1]
```diff
@@ -15,22 +16,33 @@ public final class CodeTFChangesetEntry {
   private final List<CodeTFChange> changes;
 
   private final CodeTFAiMetadata ai;
+  private final Strategy strategy;
+  private final boolean provisional;
```
Comment (gilday) [inline src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java:20]: I think the spec already calls for this to be a boolean, so I'm probably too late on this feedback, but replacing "boolean" with "enum that has two items" leaves room for expansion and better describes what this is e.g.

```java
enum Maturity { MATURE, PROVISIONAL; }
```

File: src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java [hunk 2]
```diff
@@ -15,22 +16,33 @@ public final class CodeTFChangesetEntry {
...
+      @JsonProperty("strategy") final Strategy strategy,
+      @JsonProperty("provisional") final boolean provisional,
+      @JsonProperty("fixedFindings") final List<FixedFinding> fixedFindings) {
     this.path = CodeTFValidator.requireRelativePath(path);
     this.diff = CodeTFValidator.requireNonBlank(diff);
     this.changes = CodeTFValidator.toImmutableCopyOrEmptyOnNull(changes);
     this.ai = ai;
+    this.strategy = strategy;
+    this.provisional = provisional;
+    this.fixedFindings = CodeTFValidator.toImmutableCopyOrEmptyOnNull(fixedFindings);
```
Comment (gilday) [inline src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java:40]: I assume all of these may be `null` to allow for backwards compatibility, but do we expect to eventually make these required?

File: src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java [hunk 3]
```diff
@@ -15,22 +16,33 @@ public final class CodeTFChangesetEntry {
   private final List<CodeTFChange> changes;
 
   private final CodeTFAiMetadata ai;
+  private final Strategy strategy;
+  private final boolean provisional;
+
+  /** Fixed findings that are not associated with any particular change */
```
Comment (gilday) [inline src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java:22]: Typically want to put this on the getter vs the field.

File: src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java (hunk 1 above)
Comment (drdavella) [reply src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java:20]: Noted, but at this point I think we will need to revisit it at a later date.

File: src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java (hunk 2 above)
Comment (drdavella) [reply src/main/java/io/codemodder/codetf/CodeTFChangesetEntry.java:40]: I don't have a strong opinion about this. Provisional has a reasonable default of `false` but strategy does not.

**Pull request merged**